src/heaps/binary_heap.jl

```diff
@@ -138,6 +138,7 @@ function push!{T}(h::BinaryHeap{T}, v::T)
     valtree = h.valtree
     push!(valtree, v)
     _heap_bubble_up!(h.comparer, valtree, length(valtree))
+    h
```

Comment:
Convention in `Base` seems moving to using an explicit return for functions longer than one line, but we haven't moved to that convention here yet.  So, this seems fine to me (and we can use an explicit return when someone is up to updating most or all of the return values at once.)